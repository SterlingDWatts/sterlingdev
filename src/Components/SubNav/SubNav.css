.SubNav {
  z-index: 0;
  background-color: hsl(169, 100%, 12%);
  width: 100%;
  height: 151px;
  transition-property: height;
  transition-duration: 0.25s;
  transition-timing-function: linear;
  box-shadow: 0 1px 10px 3px rgba(0, 0, 0, 0.15), 0 1px 4px 1px rgba(0, 0, 0, 0.15);
  position: relative;
}

.SubNav__container {
  display: grid;
  grid-template-columns: 27px auto;
  grid-template-rows: 1fr 1fr 1fr 1fr;
  row-gap: 16px;
  column-gap: 16px;
  font-size: 16px;
  line-height: 1;
  padding: 8px 16px 32px 16px;
  position: absolute;
  bottom: 0;
}

.SubNav--hidden {
  height: 1px;
}

.SubNav--link-name {
  text-align: left;
}

@media screen and (min-width: 750px) {
  .SubNav {
    display: none;
  }
}
