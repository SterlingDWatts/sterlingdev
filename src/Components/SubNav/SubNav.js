import React from "react";
import { Link } from "react-router-dom";
import { HashLink } from "react-router-hash-link";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faUserTie,
  faBrowser,
  faAddressCard,
  faFileUser,
} from "@fortawesome/pro-light-svg-icons";
import classnames from "classnames";
import "./SubNav.css";

function SubNav({ showSubNav, linkClick, scrollToTop }) {
  return (
    <div
      className={classnames("SubNav", {
        "SubNav--hidden": !showSubNav,
      })}
    >
      <HashLink to="/#about" onClick={linkClick}>
        <FontAwesomeIcon icon={faUserTie} />
      </HashLink>
      <HashLink to="/#about" onClick={linkClick} className="SubNav--link-name">
        About
      </HashLink>
      <HashLink to="/#projects" onClick={linkClick}>
        <FontAwesomeIcon icon={faBrowser} />
      </HashLink>
      <HashLink
        to="/#projects"
        onClick={linkClick}
        className="SubNav--link-name"
      >
        Projects
      </HashLink>
      <HashLink to="/#connect" onClick={linkClick}>
        <FontAwesomeIcon icon={faAddressCard} />
      </HashLink>
      <HashLink
        to="/#connect"
        onClick={linkClick}
        className="SubNav--link-name"
      >
        Connect
      </HashLink>
      <Link to="resume" onClick={linkClick}>
        <FontAwesomeIcon icon={faFileUser} />
      </Link>
      <Link to="resume" onClick={scrollToTop} className="SubNav--link-name">
        Resume
      </Link>
    </div>
  );
}

export default SubNav;
