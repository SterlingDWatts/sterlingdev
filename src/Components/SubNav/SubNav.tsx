import React from "react";
import { Link } from "react-router-dom";
import LightAddressCard from "../../svgs/light/LightAddressCard";
import LightBrowser from "../../svgs/light/LightBrowser";
import LightFileUser from "../../svgs/light/LightFileUser";
import LightUserTie from "../../svgs/light/LightUserTie";
import classnames from "classnames";
import "./SubNav.css";

type AppProps = {
  showSubNav: boolean;
  linkClick: () => void;
  scrollToTop: () => void;
};

const SubNav = ({ showSubNav, linkClick, scrollToTop }: AppProps) => {
  return (
    <div
      className={classnames("SubNav", {
        "SubNav--hidden": !showSubNav,
      })}
    >
      <div className="SubNav__container">
        <Link to={{ pathname: "/home", hash: "#about" }} onClick={linkClick}>
          <LightUserTie height="15px" fill="#fcd734" />
        </Link>
        <Link to={{ pathname: "/home", hash: "#about" }} onClick={linkClick} className="SubNav--link-name">
          About
        </Link>
        <Link to={{ pathname: "/home", hash: "#projects" }} onClick={linkClick}>
          <LightBrowser height="15px" fill="#fcd734" />
        </Link>
        <Link to={{ pathname: "/home", hash: "#projects" }} onClick={linkClick} className="SubNav--link-name">
          Projects
        </Link>
        <Link to={{ pathname: "/home", hash: "#connect" }} onClick={linkClick}>
          <LightAddressCard height="15px" fill="#fcd734" />
        </Link>
        <Link to={{ pathname: "/home", hash: "#connect" }} onClick={linkClick} className="SubNav--link-name">
          Connect
        </Link>
        <Link to="resume" onClick={linkClick}>
          <LightFileUser height="15px" fill="#fcd734" />
        </Link>
        <Link to="resume" onClick={scrollToTop} className="SubNav--link-name">
          Resume
        </Link>
      </div>
    </div>
  );
};

export default SubNav;
