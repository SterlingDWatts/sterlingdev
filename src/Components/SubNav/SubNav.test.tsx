import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import SubNav from "./SubNav";

describe("<SubNav />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <SubNav showSubNav={true} linkClick={() => {}} scrollToTop={() => {}} />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <SubNav showSubNav={true} linkClick={() => {}} scrollToTop={() => {}} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
