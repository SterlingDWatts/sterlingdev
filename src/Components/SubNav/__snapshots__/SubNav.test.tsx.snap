// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubNav /> renders the UI as expected 1`] = `
<div
  className="SubNav"
>
  <div
    className="SubNav__container"
  >
    <a
      href="/home#about"
      onClick={[Function]}
    >
      <svg
        height="15px"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 128C96 57.31 153.3 0 224 0C294.7 0 352 57.31 352 128C352 198.7 294.7 256 224 256C153.3 256 96 198.7 96 128zM320 128C320 74.98 277 32 224 32C170.1 32 128 74.98 128 128C128 181 170.1 224 224 224C277 224 320 181 320 128zM256 304C264.8 304 272 311.2 272 320C272 328.8 264.8 336 256 336H246.2L266.1 395.8L303.6 320.9C384.7 329.3 448 397.9 448 481.3C448 498.2 434.2 511.1 417.3 511.1H30.72C13.75 511.1 0 498.2 0 481.3C0 397.9 63.28 329.3 144.4 320.9L181.9 395.8L201.8 336H191.1C183.2 336 175.1 328.8 175.1 320C175.1 311.2 183.2 304 191.1 304H256zM224 479.1L245.9 436.2L224 370.6L202.1 436.2L224 479.1zM415.1 479.1C415.4 421.2 375.6 371.8 321.4 356.7L259.8 479.1H415.1zM188.2 479.1L126.6 356.7C72.43 371.8 32.58 421.2 32.01 479.1H188.2z"
          fill="#fcd734"
        />
      </svg>
    </a>
    <a
      className="SubNav--link-name"
      href="/home#about"
      onClick={[Function]}
    >
      About
    </a>
    <a
      href="/home#projects"
      onClick={[Function]}
    >
      <svg
        height="15px"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 96C0 60.65 28.65 32 64 32H448C483.3 32 512 60.65 512 96V416C512 451.3 483.3 480 448 480H64C28.65 480 0 451.3 0 416V96zM160 128H480V96C480 78.33 465.7 64 448 64H160V128zM128 64H64C46.33 64 32 78.33 32 96V128H128V64zM32 160V416C32 433.7 46.33 448 64 448H448C465.7 448 480 433.7 480 416V160H32z"
          fill="#fcd734"
        />
      </svg>
    </a>
    <a
      className="SubNav--link-name"
      href="/home#projects"
      onClick={[Function]}
    >
      Projects
    </a>
    <a
      href="/home#connect"
      onClick={[Function]}
    >
      <svg
        height="15px"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M480 288h-128c-8.836 0-16 7.164-16 16S343.2 320 352 320h128c8.836 0 16-7.164 16-16S488.8 288 480 288zM192 256c35.35 0 64-28.65 64-64S227.3 128 192 128S128 156.7 128 192S156.7 256 192 256zM192 160c17.64 0 32 14.36 32 32S209.6 224 192 224S160 209.6 160 192S174.4 160 192 160zM224 288H160c-44.18 0-80 35.82-80 80C80 376.8 87.16 384 96 384s16-7.164 16-16C112 341.5 133.5 320 160 320h64c26.51 0 48 21.49 48 48c0 8.836 7.164 16 16 16s16-7.164 16-16C304 323.8 268.2 288 224 288zM512 32H64C28.65 32 0 60.65 0 96v320c0 35.35 28.65 64 64 64h448c35.35 0 64-28.65 64-64V96C576 60.65 547.3 32 512 32zM544 416c0 17.64-14.36 32-32 32H64c-17.64 0-32-14.36-32-32V96c0-17.64 14.36-32 32-32h448c17.64 0 32 14.36 32 32V416zM480 224h-128c-8.836 0-16 7.164-16 16S343.2 256 352 256h128c8.836 0 16-7.164 16-16S488.8 224 480 224zM480 160h-128c-8.836 0-16 7.164-16 16S343.2 192 352 192h128c8.836 0 16-7.164 16-16S488.8 160 480 160z"
          fill="#fcd734"
        />
      </svg>
    </a>
    <a
      className="SubNav--link-name"
      href="/home#connect"
      onClick={[Function]}
    >
      Connect
    </a>
    <a
      href="/resume"
      onClick={[Function]}
    >
      <svg
        height="15px"
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M365.3 125.3l-106.5-106.5C246.7 6.742 230.5 0 213.5 0H64C28.65 0 0 28.65 0 64l.0065 384c0 35.35 28.65 64 64 64H320c35.35 0 64-28.65 64-64V170.5C384 153.5 377.3 137.3 365.3 125.3zM224 34.08c4.477 1.566 8.666 3.846 12.12 7.299l106.5 106.5C346.1 151.3 348.4 155.5 349.9 160H240C231.2 160 224 152.8 224 144V34.08zM272 480h-160v-16C112 437.5 133.5 416 160 416h64c26.51 0 48 21.49 48 48V480zM352 448c0 17.64-14.36 32-32 32h-16v-16c0-44.18-35.82-80-80-80H160c-44.18 0-80 35.82-80 80V480H64c-17.64 0-32-14.36-32-32V64c0-17.64 14.36-32 32-32h128v112C192 170.5 213.5 192 240 192H352V448zM192 224C156.7 224 128 252.7 128 288s28.65 64 64 64s64-28.65 64-64S227.3 224 192 224zM192 320c-17.64 0-32-14.36-32-32s14.36-32 32-32s32 14.36 32 32S209.6 320 192 320z"
          fill="#fcd734"
        />
      </svg>
    </a>
    <a
      className="SubNav--link-name"
      href="/resume"
      onClick={[Function]}
    >
      Resume
    </a>
  </div>
</div>
`;
